app/src/pages/eligibility.tsx

```diff
@@ -9,73 +8,104 @@ import ButtonLink from '@components/ButtonLink'
-      setContinueBtn({
-        labelKey: 'updateAndReturn',
-        route: previousRoute,
-      })
-    } else setContinueBtn({ ...continueBtn, route: incomeRoute })
-  }, [form.none, props.previousRoute])
+    setForm(session.eligibility)
+  }, [session.eligibility])
 
+  // Set up action button and routing.
+  const actionButtonLabel = reviewMode ? 'updateAndReturn' : 'continue'
+
+  // Set a state for whether the form requirements have been met and the
+  // form can be submitted. Otherwise, disable the submit button.
+  const [disabled, setDisabled] = useState(true)
+  // Use useEffect() to properly load the data from session storage during react hydration
+  // Since we need to use useEffect to update the enabled/disabled state for the button,
+  // this also handles anytime the form state is updated, so we don't need to call
+  // setDisabled during handleChange()
   useEffect(() => {
-    setDisabled(!requiredMet())
+    setDisabled(!isValidEligibility(form))
   }, [form])
 
+  // Handle all form element changes.
+  // - Determine new form values
+  // - Update the form and session states
+  // - Update the action button and routing for which page the user should be
+  //   routed to based on user-entered data
   const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
-    const { value, name }: { value: string; name: string } = e.target
-    const castValue = value as keyof typeof form
-    let newValue
+    const { value, name, type }: { value: string; name: string; type: string } =
+      e.target
+    let newForm
 
-    if (['residential', 'before'].includes(name)) {
-      newValue = { [name]: value }
-    } else {
-      newValue = { [castValue]: !form[castValue] }
+    // Handle radio buttons and checkboxes need to be handled differently.
+    // Radio buttons are a boolean setting, so override the existing value.
+    if (type === 'radio') {
+      newForm = { ...form, [name]: value }
     }
+    // Checkboxes are an array because multiple options are accepted,
+    // so manage the array elements.
+    else if (type === 'checkbox') {
+      // Cast the name as a key in form.
+      const castName = name as 'categorical' | 'adjunctive'
+      // If the checkbox is checked and that checkbox value isn't in the array,
+      // add it to the array.
+      // @TODO: consider using Immer
+      if (e.target.checked && !form[castName].includes(value)) {
+        const checkboxArray = [...form[castName]]
+        checkboxArray.push(value)
+        newForm = { ...form, [castName]: checkboxArray }
 
-    const newForm = { ...form, ...newValue }
+        // If the checkbox is "None of the above", uncheck all other values.
+        if (value === 'none') {
+          newForm = { ...form, [castName]: [value] }
+        }
+        // Otherwise, make sure "None of the above" is unchecked.
+        else {
+          newForm[castName] = newForm[castName].filter((element) => {
+            return element !== 'none'
+          })
+        }
```

Comment:
Notes for reviewers: This handles the "none of the above" checkbox logic:

- When you check "none of the above", all other checkboxes in that question should become unchecked
- When "none of the above" is checked and then you check a different checkbox, "none of the above" should become unchecked